### ISSUE TITLE
[Test]: Fix doom loop detector false positives (2 failures)

### ISSUE THREAD

**Issue by kesslerio** [kesslerio/vibe-check-mcp#251]
## Problem
2 tests in TestRealWorldScenarios are incorrectly flagging healthy content as doom loops:
- `test_healthy_implementation_focused_content` - Returns 'doom_loop_detected' instead of 'healthy_analysis'
- `test_productive_decision_making` - Returns 'doom_loop_detected' instead of 'healthy_analysis'

## Background
Already attempted fix by adding `setup_method()` to reset session tracking, but tests still failing. The detection thresholds appear too sensitive, causing false positives on legitimate productive content.

## Files Affected
- `tests/unit/test_doom_loop_detector.py::TestRealWorldScenarios`
- `src/vibe_check/patterns/doom_loop_detector.py` (lines 267-277 - thresholds)

### Healthy Implementation (should NOT trigger):
```python
I'm implementing user authentication today.
Step 1: Set up OAuth with Google (using their official SDK)
Step 2: Create user session management
...
Timeline: Complete by end of week, iterate based on user feedback.
```

### Productive Decision Making (should NOT trigger):
```python
After evaluating options, I'm going with PostgreSQL.
Reasons: Team expertise, ACID compliance, good performance.
Next steps: [concrete action items]
```

## Root Cause
The doom loop detection appears to be too aggressive in its pattern matching, possibly triggering on:
- Discussion of technology choices (even when decision is made)
- Structured planning content (mistaken for analysis paralysis)
- Timeline mentions (mistaken for indefinite postponement)

## Possible Fixes
1. **Adjust detection thresholds** in `doom_loop_detector.py` (lines 267-277)
   - Increase similarity threshold from current value
   - Adjust minimum iterations before triggering
   
2. **Refine pattern detection logic**
   - Better distinguish between healthy planning and analysis paralysis
   - Recognize concrete action items as positive signals
   
3. **Update test content** to be more clearly action-focused (if needed)

## Acceptance Criteria
- [ ] `test_healthy_implementation_focused_content` passes
- [ ] `test_productive_decision_making` passes
- [ ] Both tests correctly identify content as 'healthy_analysis'
- [ ] Existing doom loop detection tests still pass (no regression)
- [ ] Detection correctly identifies actual doom loops (verify with existing passing tests)

## Testing
```bash
# Run specific failing tests
export PYTHONPATH=src:. && pytest tests/unit/test_doom_loop_detector.py::TestRealWorldScenarios::test_healthy_implementation_focused_content -v
export PYTHONPATH=src:. && pytest tests/unit/test_doom_loop_detector.py::TestRealWorldScenarios::test_productive_decision_making -v

# Run all doom loop detector tests
export PYTHONPATH=src:. && pytest tests/unit/test_doom_loop_detector.py -v

# Verify no regression in pattern detection
export PYTHONPATH=src:. && pytest tests/unit/ -k "doom_loop" -v
```

## Related Issues
- Parent issue: #245 (Pre-v0.6.0 test failures)
- Working branch: `fix/test-failures-pre-v0.6.0`

## Technical Notes
- Session tracking reset was already attempted via `setup_method()` 
- Issue is likely in the detection logic, not test setup
- Consider reviewing similar false positive patterns in production usage

## Priority
P2 - Part of v0.6.0 release test suite fixes

## Estimated Time
30-45 minutes

**Comment by kesslerio** [comment]
Fixed by PR #257 - Added positive progress signal detection to reduce false positives when text includes clear action plans and decisions
